File: 回文链表/C++/original.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 * 思路和python一样
 */
class Solution {
public:
    bool isPalindrome(ListNode* head) {
        if(head==NULL || head->next == NULL ) return true;
        vector<int> temp;
        int n=0;
        while(head){
            n += 1;
            temp.push_back(head->val);
            head = head->next;
        }
        for(int i=0; i<n; ++i){
            if(temp[i] != temp[n-i-1]) return false;
        }
        return true;
    }
};

File: 回文链表/python3/original.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def isPalindrome(self, head):
        """
        :type head: ListNode
        :rtype: bool
        思路：i计算链表中节点数量，dic存放链表的节点序号
        和节点数值，然后遍历
        """
        i = 0
        dic = {}
        while head:
            i += 1
            dic[i] = head.val
            head = head.next
        for j in range(1, int(i/2)+1):
            if dic[j] != dic[i-j+1]:
                return False
        return True